src/status/votes.py

```diff
@@ -0,0 +1,89 @@
+from src.decorators import event_listener
+from src.containers import UserDict
+from src.functions import get_players
+from src.events import Event
+
+__all__ = ["force_vote", "force_abstain", "can_vote", "can_abstain"]
+
+# FORCED_COUNTS is incremented whenever we're forcing someone to vote for someone else
+# A positive number indicates that their vote is being forced towards any number of targets
+# whereas a negative number indicates that their vote is being forced towards abstention.
+# 0 indicates not being forced.
+# FORCED_TARGETS is used to see which targets are being force-voted for.
+# If multiple calls to force_vote() force different targets, the union of all of those targets is taken.
+
+FORCED_COUNTS = UserDict() # type: UserDict[users.User, int]
+FORCED_TARGETS = UserDict() # type: UserDict[users.User, UserSet]
+
+def force_vote(var, votee, targets):
+	evt = Event("force_vote", {})
+	if not evt.dispatch(var, votee, targets):
+		return
+
+	FORCED_COUNTS[votee] = FORCED_COUNTS.get(votee, 0) + 1
+	FORCED_TARGETS.get(votee, UserSet()).update(targets)
+	if FORCED_COUNTS[votee] == 0:
+		# don't clear out FORCED_TARGETS, in case a future call re-forces votes
+		# we want to maintain the full set of people to vote for
+		del FORCED_COUNTS[votee]
+
+def force_abstain(var, votee):
+	evt = Event("force_vote", {})
+	if not evt.dispatch(var, votee, None):
+		return
+
+	FORCED_COUNTS[votee] = FORCED_COUNTS.get(votee, 0) - 1
+	if FORCED_COUNTS[votee] == 0:
+		del FORCED_COUNTS[votee]
+
```

Comment:
Merge `force_vote` and `force_abstain` into a single `add_weight` (or `_add_weight`) function, then have both `force_vote` and `force_abstain` call it with proper parameters.